multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -47,16 +73,317 @@ type ResourceExportReconciler struct {
+	resImport := &mcsv1alpha1.ResourceImport{}
+	switch resExport.Spec.Kind {
+	case common.ServiceKind:
+		resImport, changed, err = r.refreshServiceResourceImport(&resExport, existResImport, createResImport)
+	case common.EndpointsKind:
+		resImport, changed, err = r.refreshEndpointsResourceImport(&resExport, existResImport, createResImport)
+	}
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if createResImport {
+		if err = r.Client.Create(ctx, resImport, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "fail to create ResourceImport", "resourceimport", resImportName.String())
+			return ctrl.Result{}, err
+		}
+		klog.InfoS("create ResourceImport successfully", "resourceimport", resImportName.String())
+	}
+
+	if changed && !createResImport {
+		klog.InfoS("update ResourceImport for ResoureExport", "resourceimport", resImportName.String(), "resourceexport", req.NamespacedName)
+		if err = r.handleUpdateEvent(ctx, resImportName, resImport, &resExport); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
 	return ctrl.Result{}, nil
 }
 
+func (r *ResourceExportReconciler) handleUpdateEvent(ctx context.Context, resImpName types.NamespacedName,
+	resImport *mcsv1alpha1.ResourceImport, resExport *mcsv1alpha1.ResourceExport) error {
+	var err error
+	if err = r.Client.Update(ctx, resImport, &client.UpdateOptions{}); err != nil {
+		klog.ErrorS(err, "fail to update ResourceImport", "resourceimport", resImpName.String())
+		return err
+	}
+	latestResImport := &mcsv1alpha1.ResourceImport{}
+	err = r.Client.Get(ctx, resImpName, latestResImport)
+	if err != nil {
+		klog.ErrorS(err, "fail to get latest ResourceImport", "resourceimport", resImpName.String())
+		return err
+	}
+
+	newStatus := mcsv1alpha1.ResourceImportClusterStatus{
+		ClusterID: resExport.Labels["sourceClusterID"],
+		Conditions: []mcsv1alpha1.ResourceImportCondition{{
+			Type:               mcsv1alpha1.ResourceImportSucceeded,
+			Status:             corev1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            "update is successful",
+		}},
+	}
+	var matchedStatus bool
+	for k, s := range latestResImport.Status.ClusterStatuses {
+		if s.ClusterID == resExport.Labels["sourceClusterID"] {
+			matchedStatus = true
+			latestResImport.Status.ClusterStatuses[k] = newStatus
+		}
+	}
+	if !matchedStatus {
+		latestResImport.Status.ClusterStatuses = append(latestResImport.Status.ClusterStatuses, newStatus)
+	}
+	if err := r.Client.Status().Update(ctx, latestResImport, &client.UpdateOptions{}); err != nil {
+		klog.ErrorS(err, "fail to update ResourceImport Status", "resourceimport", resImpName.String())
+		return err
+	}
+	return nil
+}
+
+func (r *ResourceExportReconciler) handleDeleteEvent(ctx context.Context, resImportName types.NamespacedName, resExport *mcsv1alpha1.ResourceExport) error {
+	reList := &mcsv1alpha1.ResourceExportList{}
+	err := r.Client.List(ctx, reList, &client.ListOptions{LabelSelector: getLabelSelector(resExport)})
```

Comment:
because multiple resExports from different member clusters will be converged into one ResourceImport, when there is a ResourceExport deletion happens, it might just need to update ResourceImport instead of delete it, so we need to list to check if we need to delete or just update ResourceImport.

Reply:
Ok, thought we can just remove endpoints from the ResourceImport. Not sure that is a better strategy or not.

But could you add some comments?

Reply:
done, add some comments for this method, if we remove Endpoints from ResourceImport, we need to comparing Endpoint to the whole Subset list in ResourceImport and remove the target one, I feel it's hard and more expensive comparing with simply append existing ResourceExports to a new ResourceImport, so I choose this way.